packages/components-vue/projects/vue-wrapper/tests/unit/specs/utils.spec.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  addEventListenerToElementRef,
+  getPrefixedTagName,
+  prefixInjectionKey,
+  syncProperties,
+} from '../../../src/utils';
+import * as Vue from 'vue';
+
+describe('getPrefixedTagName()', () => {
+  it('should call inject() with correctParameters', () => {
+    const spy = jest.spyOn(Vue, 'inject').mockReturnValue('');
+
+    getPrefixedTagName('p-text');
+
+    expect(spy).toBeCalledWith(prefixInjectionKey);
+  });
+
+  it('should call throw error if prefix is undefined', () => {
+    jest.spyOn(Vue, 'inject').mockReturnValue(undefined);
+
+    let error;
+    try {
+      getPrefixedTagName('p-text');
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toBeDefined();
+  });
+
+  it('should return tagName if no prefix is defined', () => {
+    jest.spyOn(Vue, 'inject').mockReturnValue('');
+    const tagName = 'p-text';
+    getPrefixedTagName('p-text');
+
+    expect(getPrefixedTagName(tagName)).toBe(tagName);
+  });
+
+  it('should return prefixed tagName if prefix is defined', () => {
+    const prefix = 'my-prefix';
+    jest.spyOn(Vue, 'inject').mockReturnValue(prefix);
+    const tagName = 'p-text';
+
+    expect(getPrefixedTagName(tagName)).toBe(prefix + '-' + tagName);
+
+    const prefix2 = 'another-prefix'
+    jest.spyOn(Vue, 'inject').mockReturnValue(prefix2);
+
+    expect(getPrefixedTagName(tagName)).toBe(prefix2 + '-' + tagName);
+  });
+});
+
+describe('syncProperties()', () => {
+  it('should sync passed properties with passed element', () => {
+    const props: {
+      customProp1: string;
+      customProp2: boolean;
+      customProp3: number;
+      customProp4: {};
+    } & Partial<HTMLElement> = { customProp1: 'some prop', customProp2: true, customProp3: 1, customProp4: {} };
```

Comment:
Partial type can be removed I think.

Reply:
It is needed due the typing of syncProperties, where passed props have the type `Partial<HTMLElement>`